### PR TITLE
Core/Spells: Fix mage Flame on talent

### LINKED ISSUE #19583: Core/Script The talent Flame On has an issue.
**Description:**

The talent Flame On(ID:205029) given to Mage Fire at lv.60 has an issue

Flame On (passive): Reduces the cooldown of Fire Blast by 2 secconds and increases the maximum number of charges by 1

**Current behaviour:**

When you choose this talent it increase 1 charge of Fire Blast, but it doesn´t reduces the cooldown

**Expected behaviour:**

When you choose this talent it should reduces the cooldown of Fire Blast by 2 secconds and increases the maximum number of charges by 1

**Steps to reproduce the problem:**

1. Create a Mage character
2. Reach lv.59, choose the Fire Spec
3. Check the cooldown of the spell Fire Blast
4. Reach lv.60 and choose the talent Flame On
5. Check the cooldown of the spell Fire Blast

**Branch(es):**

Master

**TC rev. hash/commit:** 

7c66e7bbc7a401e7a2d8f2bdcead6531678262ea

**TDB version:**  7.2.0+updates

**Operating system:** Windows 10

Description in wowhead http://www.wowhead.com/spell=205029/flame-on

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix talent Flame On

**Issues addressed:**

Closes #19583

**Tests performed:**

Tested in-game

### CODE REVIEW

**Comment by aquadeus** [comment]
Idk what blizzard is doing here, for a 100% correct fix we need a float value here

**Comment by mdX7** [comment]
Blizz indeed is using float values, our systems gotta be adjusted to use floats someday

File: src/server/scripts/Spells/spell_mage.cpp [hunk 1]
```diff
@@ -564,6 +564,23 @@ class spell_mage_firestarter_dots : public AuraScript
...
 
+// 205029 - Flame On
+class spell_mage_flame_on : public AuraScript
+{
+   PrepareAuraScript(spell_mage_flame_on);
+
+   void CalculateAmount(AuraEffect const* /*aurEff*/, int32& amount, bool& canBeRecalculated)
+   {
+       canBeRecalculated = false;
+       amount = -16;
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_mage.cpp:575]: if its just set to a constant we likely may use SpellMgr for this. Also we may calculate it directly `effect3 basepoints / rechargeTime * 100`

Comment (Shauren) [reply src/server/scripts/Spells/spell_mage.cpp:575]: Eeeh its more complicated than that.
Cooldown of Fire Blast is reduced by haste and the talent is then supposed to reduce it by flat 2 seconds (that means this aura is supposed to update value with stats (haste) changes)

Comment (mdX7) [reply src/server/scripts/Spells/spell_mage.cpp:575]: shouldn't we already take that into account by using the current rechargetime for the calculation?

Comment (Shauren) [reply src/server/scripts/Spells/spell_mage.cpp:575]: no, this is fine

**Pull request merged**